packages/block-library/src/paragraph/index.js

```diff
@@ -30,18 +30,21 @@ export const settings = {
 	__experimentalLabel( attributes, { context } ) {
 		const customName = attributes?.metadata?.name;
 
-		if ( context === 'list-view' && customName ) {
-			return customName;
-		}
-
 		if ( context === 'accessibility' ) {
-			if ( customName ) {
+			if (
+				customName &&
+				! attributes?.metadata?.bindings?.source === 'pattern/overrides'
+			) {
```

Comment:
I don't like the hardcoding of `pattern/overrides` here. I also don't like that the pattern special case would need to be accounted for in every relevant block. Should we explore an alternative?

Reply:
I think the name is wrong ('core/pattern-overrides').

----

On a different note, I'm wondering whether this is a pattern override specific issue. The problem seems to be that when a paragraph is named, the name is announced instead of the content. It is perhaps worse for pattern overrides since all editable blocks will be named, but the problem still exists for regular paragraphs too.

Either way it seems like screenreader users might miss valuable context (if the name isn't announced a user might wonder why given they explicitly gave it the name).

A compromize could be to announce both the name and the content if it can be made clear enough.

Reply:
> I think the name is wrong ('core/pattern-overrides').

`pattern/overrides` is how it's defined in the block context, so that's what I was just following that pattern:

https://github.com/WordPress/gutenberg/blob/b0c789282e8c43240edd089d5661d1146cd211a0/lib/compat/wordpress-6.6/block-bindings/pattern-overrides.php#L52-L59

https://github.com/WordPress/gutenberg/blob/b0c789282e8c43240edd089d5661d1146cd211a0/packages/block-editor/src/hooks/use-bindings-attributes.js#L183


> On a different note, I'm wondering whether this is a pattern override specific issue. The problem seems to be that when a paragraph is named, the name is announced instead of the content. It is perhaps worse for pattern overrides since all editable blocks will be named, but the problem still exists for regular paragraphs too.
>
> Either way it seems like screenreader users might miss valuable context (if the name isn't announced a user might wonder why given they explicitly gave it the name).
>
> A compromize could be to announce both the name and the content if it can be made clear enough.

This is a good point. I think announcing both the name and content is the right approach. I can implement that change as part of this PR if we decide to move forward with it, or create an issue and perhaps handle in a separate PR if not.